drivers/i2c/i2c_andes_atciic100.c

```diff
@@ -44,7 +44,8 @@ static void i2c_atciic100_default_control(const struct device *dev)
 	struct i2c_atciic100_dev_data_t *dev_data = dev->data;
 	uint32_t reg = 0;
 
-	k_sem_init(&dev_data->i2c_busy_sem, 1, 1);
+	k_sem_init(&dev_data->bus_lock, 1, 1);
```

Comment:
You initialize the `bus_lock` as full, but then also perform a `k_sem_give` in `i2c_atciic100_configure()`.

You probably want to ether initialize `bus_lock` as empty or remove the redundant `k_sem_give` from the configure routine.